SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKTestGuidelines.h

```diff
@@ -0,0 +1,56 @@
+/*
+0. How to run tests:
+	- Tests can be run via Session Frontend in UE4 Editor
+
+	- Tests can be run via command line
+
+1. Folder structure
+	- Tests are located in a separate SpatialGDKTests module.
+	So for every component that needs to be tested - it has to be exposed via corresponding macro.
+	E.g. `class FRPCContainer` -> `class SPATIALGDK_API FRPCContainer`, to make FRPCContainer testable.
+	WThe macro is different in each module SPATIALGDK_API, SPATIALGDKSERVICES_API etc.
+
+	- The folder structure inside SpatialGDKTests should resemble the folder structure for components being tested.
+	E.g. If file tested is
+	`UnrealGDK\SpatialGDK\Source\SpatialGDK\Public\Utils\RPCContainer.cpp`,
+	then the corresponding test should be located in
+	`UnrealGDK\SpatialGDK\Source\SpatialGDKTests\SpatialGDK\Utils\RPCContainer\RPCContainerTest.cpp`.
+	There should be a folder with the same name as component tested. It should include the test and all the supporting files.
+
+2. Test definitions (check TestDefinitions.h for more info)
+    - We have defined 3 types of Macro to be used when writing tests:
+	(https://docs.unrealengine.com/en-US/Programming/Automation/TechnicalGuide/index.html has more information)
+	TEST - the most common simple test
+	COMPLEX_TEST - same as simple test, but allows having multiple test cases run through the same test function body.
+	DEFINE_LATENT_COMMAND... - used to run tests that require more than 1 thread.
+
+	- There are 5 types of mock objects we can use in tests:
+	Dummy objects
+		are passed around but never actually used. Usually they are just used to fill parameter lists.
+
+	Fake objects
+		actually have working implementations, but usually take some shortcut which makes them not suitable for production (an InMemoryTestDatabase is a good example).
+
+	Stubs
+		provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
+
+	Spies
+		are stubs that also record some information based on how they were called.
+		One form of this might be an email service that records how many messages it was sent.
+
+	Mocks
+		are pre-programmed with expectations which form a specification of the calls they are expected to receive.
+		They can throw an exception if they receive a call they don't expect and are checked during verification to ensure they got all the calls they were expecting.
+
+3. Test naming convention
+	- `GIVEN_WHEN_THEN` should be used.
+	E.g. `GIVEN_one_and_two_WHEN_summed_THEN_the_sum_is_three`
+	Tests should contain only one TestTrue/TestFalse statement.
```

Comment:
That can be quite limiting. Are we sure we want this?

Reply:
Especially when there are latent commands

Reply:
It is quite limiting, I'm still on the fence about whether to restrict us to just one test statement.

Reply:
Updated to `Should contain a minimal number of TestTrue/TestFalse statement`